sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -2563,7 +2563,23 @@ class Analyzer(override val catalogManager: CatalogManager)
 
       case Sort(sortOrder, global, agg: Aggregate) if agg.resolved =>
         // We should resolve the references normally based on child (agg.output) first.
-        val maybeResolved = sortOrder.map(_.child).map(resolveExpressionByPlanOutput(_, agg))
```

Comment:
I think this bug is much more complicated. Think about `order by sum(id)` and `order by abs(id)`. For the first case, we want to resolve `id` to the table column and push `sum(id)` to the Aggregate. For the second case, we want to resolve `id` to `sum(id) as id`.

How to define a clear rule here?

Reply:
My proposal is: we still try to resolve the column to `agg.output` first, but temporarily (use `TempResolvedColumn`). If the order by expression is resolved but contains aggregate functions, unresolve the column and call `resolveOperatorWithAggregate`.

Reply:
e.g.
```
val maybeResolved = sortOrder.map(_.child).map { expr =>
  val resolved = resolveTemp(expr, agg)
  if (resolved.exists(_.isInstanceOf[AggregateFunction])) {
    unresolve(resolved)
  } else {
    resolved
  }
}
resolveOperatorWithAggregate...
```